DatabaseWorkshop.md

```diff
@@ -0,0 +1,185 @@
+
+### Step 1 - Create a Database
+Create a new database for the project called `tododb` using your Terminal:
+```
+createdb tododb
+```
+
+### Step 2 - Updating Package.swift
+Your `Package.swift` needs two new Packages for the PostgreSQL and ORM to work. 
+1. Open the `ToDoServer` > `Package.swift` file
+2. Add the following two lines to the end of the Package.swift's dependencies section:
+```swift
+.package(url: "https://github.com/IBM-Swift/Swift-Kuery-ORM", from: "0.3.1"),
+.package(url: "https://github.com/IBM-Swift/Swift-Kuery-PostgreSQL", from: "1.2.0"),
+```
+3. Declare these new packages in the "Application" target (note the lack of hyphens):
+```swift
+.target(name: "Application", dependencies: ["SwiftKueryPostgreSQL", "SwiftKueryORM", "KituraCORS", "KituraOpenAPI", "Kitura", "CloudEnvironment","SwiftMetrics","Health",]),
+```
+4. We need to fetch the new dependencies and generate a new Xcode Project file. To do so, make sure Xcode is closed and:
+```bash
+cd ~/ToDoBackend/ToDoServer
+swift package generate-xcodeproj
+open ToDoServer.xcodeproj
+```
+
+### Step 3 - Importing into the Application
+1. In Xcode, open `Sources` > `Application` > `Application.swift`
+2. Add imports for the two new libraries below the other import statements at the top of the file:
+```swift
+import SwiftKueryORM
+import SwiftKueryPostgreSQL
+```
+### Step 4 - Extending ToDo to conform to Model
+In order for your `ToDo` struct located in `Application` > `Model.swift`, we need to conform `ToDo` to Swift-Kuery-ORMs Model protocol. At the bottom of `Application.swift` add:
+```swift
+extension ToDo: Model {
+}
+```
+This exposes new methods to ToDo for saving, finding, deleting etc. from a database.
+
+### Step 5 - Creating a connection with the database
+Add the following Class to the bottom of the file, outside the `App` classes final curly brace:
+```swift
+class Persistence {
+	static func setUp() {
+		let pool = PostgreSQLConnection.createPool(host: "localhost", port: 5432, options: [.databaseName("tododb")], poolOptions: ConnectionPoolOptions(initialCapacity: 10, maxCapacity: 50, timeout: 10000))
+		Database.default = Database(pool)
+	}
+}
+```
+This code is going to create a connection pool and set it as the default database. We now need to call the `setUp()` method inside  `App`s `postInit()` method:
+```swift
+		Persistence.setUp()
+		do {
+			try ToDo.createTableSync()
+		} catch let error {
+			print("Database already exists.")
+		}
+```
+This will create a table for us. We can now begin modifying the methods to switch saving to the `todoStore` and instead save to the database.
+
+To test everything is working as it should, we can run the Kitura server and check that a table called `ToDos` is created:
+```sql
+psql tododb
+SELECT * FROM "ToDos"
+```
+This should print the columns of a table, with no data in any of the colums. We will change the methods for handling requests now to store and retrieve data from this database.
```

Comment:
This should print the column names of the `ToDos` table with no data in (i.e. no rows).

Now we will start modifying the methods which handle requests to stop using the `todoStore` array and instead to store and retrieve data from this database table.

Reply:
✅